website/content/en/highlights/2022-02-08-disk-buffer-v2-beta.md

```diff
@@ -0,0 +1,81 @@
+buffers lack the ability to provide _data durability._ If any events are still in an in-memory
+buffer and Vector experiences an error that terminates the process, or the system running Vector
+itself crashes, those buffered events would be permanently lost.  As an alternative to in-memory
+buffers, we provide disk buffers, which allow writing the events to disk, providing durability and
+persistence, regardless of issues with Vector and the host system.
+
+## Current disk buffer implementation
+
+The initial disk buffer implementation in Vector was based on [LevelDB][leveldb]. LevelDB can unquestionably meet the "durability" guarantees that Vector wants to provide, but falls short at providing _consistent_ performance.  LevelDB's design, to write to many files, and then eventually merge them back together in the background, is overkill for our needs: Vector only ever writes events to its buffer in a sequential fashion, and paying the cost of this merging and write amplification is a cost that we'd rather not pay.  It not only reduces the consistency of Vector's performance, but can cause issues with resource consumption as well.
+
+As an example, we currently use a forked version of LevelDB because, by default, LevelDB might load up to 1000 files, via mmap, into a Vector process.  Having to figure this out _after_ a user experiences a confusing out-of-memory process crash is not fun for us or our users.  As well, integrating LevelDB into Vector, is suboptimal on two fronts: integrating a C/C++ dependency involves multiple crates and build script tweaks for ensuring builds work on different platforms, and marrying the synchronous design of LevelDB with the asynchronous design of Vector is tricky at best.
+
+## Redesigned disk buffer implementation
+
+In order to address these issues, and more, we've written a new disk buffer implementation that is better suited for Vector's specific needs.  At a high level, the new disk buffer implementation works more like an actual log -- files being written once, and read sequentially -- and not at all like a database. This means we perform no LevelDB-specific operations such as compaction, and in turn, we don't pay the additional cost of doing so.  Our design was built from the ground up to fit within Vector rather than having to be molded into something that would work.  This has enabled the new disk buffer to provide more consistent throughput and latency, as well as memory and CPU consumption.
+
+## The road to a stable release
+
+While we're excited to have you try out the new disk buffer for your own workloads, it is still a
+**beta** feature: in general, there are no known issues, but you may encounter an issue that could cause data loss or
+cause Vector to become unresponsive. We're working hard to continue testing and hardening the new
+disk buffer implementation for a planned stable release in 0.21.
+
+## Trying out the new disk buffer implementation today
+
```

Comment:
I'd add something here at the top of this section encouraging users to try it out and give us feedback.

Reply:
Done.